app/Models/User.php

```diff
@@ -48,4 +47,48 @@ public function posts()
     {
         return $this->hasMany(Post::class);
     }
+
+    public function relations()
+    {
+        return $this->hasMany(Relation::class, 'user_id');
+    }
+
+    public function relationsFriend()
+    {
+        return $this->hasMany(Relation::class, 'friend_id');
+    }
+
+    /**
+     * Check user is exist in db or not
+     *
+     */
+    public function isExistUser($id)
+    {
+        return self::find($id);
+    }
+
+    /**
+     * Check user is able to add friend or not
+     *
+     */
+    public function isFriendable($id)
+    {
+        $user = self::isExistUser($id);
+        if ($user != null) {
+            if ($user->is_added != 1 || $user->id == $this->id) {
+                return false;
+            }
+            return true;
+        }
+        return false;
+    }
```

Comment:
Ko viết các kiểu if xxxx thì return true/false.

```suggestion
        return $user != null && $user->is_added == 1 && $user->id != $this->id
    }
```